react-carousel-component/my-app/src/Carousel.tsx

```diff
@@ -16,12 +15,13 @@ export default function Carousel({ images }: Props) {
   }
 
   function handleNextClick() {
-    setActiveIndex((activeIndex + 1 + images.length) % images.length);
+    setActiveIndex((activeIndex + 1) % images.length);
   }
 
   useEffect(() => {
-    setInterval(handleNextClick, 3000);
-  });
+    const interval = setInterval(handleNextClick, 3000);
+    return clearInterval(interval);
+  }, []);
 
```

Comment:
Don't you get a lint warning that you are missing a dependency (handleNextClick)?
You need to fix that. You probably want to use `useCallback` to fix it.